Sources/SwiftDocCUtilities/ArgumentParsing/ActionExtensions/ConvertAction+CommandInitialization.swift

```diff
@@ -100,3 +100,8 @@ private func symbolGraphFiles(in directory: URL?) -> [URL] {
     return subpaths.map { directory.appendingPathComponent($0) }
         .filter { DocumentationBundleFileTypes.isSymbolGraphFile($0) }
 }
+
+private protocol _DeprecatedSymbolGraphFilesAccess {
+    var additionalSymbolGraphFiles: [URL] { get }
+}
+extension Docc.Convert: _DeprecatedSymbolGraphFilesAccess {}
```

Comment:
And, how does this actually work? By extending Docc.Convert so it conforms to a private protocol, are we preventing any other code from using the attributes (`additionalSymbolGraphFiles`) of this private protocol outside of this file?

Reply:
`Docc.Convert` already has an `additionalSymbolGraphFiles ` property but it's marked as deprecated [here](https://github.com/apple/swift-docc/pull/806). Since it's deprecated, any code that accesses it will result in a deprecation warning, even within the same module. 

That leaves us with 3 choices:
- Do nothing and live with the added depreciation warnings for a couple of releases.
- Stop accessing the deprecated API property, breaking the behavior for any client who passes this flag
- Write a bit of extra code to be able to avoid the deprecation warning.

I opted for the third choice. The way this private protocol avoids the deprecation warning works in 3 steps:

 - ```swift
    private protocol _DeprecatedSymbolGraphFilesAccess {
        var additionalSymbolGraphFiles: [URL] { get }
    }
    ```
    defines a protocol with a property with the same name and type as the deprecated API. This property of this protocol isn't deprecated. This protocol is only available within this source file.  
 - ```swift
    extension Docc.Convert: _DeprecatedSymbolGraphFilesAccess {}
    ```
    conforms the `Convert` command to this private protocol. Since `Convert` already has a property with this name and type it doesn't need to implement anything new.
 - ```swift
    (convert as _DeprecatedSymbolGraphFilesAccess).additionalSymbolGraphFiles
    ```
    casts the convert value to the deprecated property; thus shadowing the deprecated `Convert.additionalSymbolGraphFiles` with the not deprecated `_DeprecatedSymbolGraphFilesAccess.additionalSymbolGraphFiles`

It looks rather strange but it allows us to internally call the deprecated API without warnings about it being deprecated. I know that there aren't other places who access `Convert.additionalSymbolGraphFiles` because those would result in deprecation warnings. If any new code called `Convert.additionalSymbolGraphFiles` it would work but result in a deprecation warning.